components/shapediver/viewport/ViewportHistoryButtons.tsx

```diff
@@ -0,0 +1,260 @@
+import ImportModelStateDialog from "@AppBuilderShared/components/shapediver/ui/ImportModelStateDialog";
+import Icon from "@AppBuilderShared/components/ui/Icon";
+import TooltipWrapper from "@AppBuilderShared/components/ui/TooltipWrapper";
+import {useParameterImportExport} from "@AppBuilderShared/hooks/shapediver/parameters/useParameterImportExport";
+import {useModelState} from "@AppBuilderShared/hooks/shapediver/useModelState";
+import {useViewportHistory} from "@AppBuilderShared/hooks/shapediver/viewer/useViewportHistory";
+import {useShapeDiverStoreParameters} from "@AppBuilderShared/store/useShapeDiverStoreParameters";
+import {IconTypeEnum} from "@AppBuilderShared/types/shapediver/icons";
+import {IParameterChanges} from "@AppBuilderShared/types/store/shapediverStoreParameters";
+import {ActionIcon, Box, MantineStyleProp, Menu} from "@mantine/core";
+import React, {useCallback, useMemo, useState} from "react";
+
+interface Props {
+	/**
+	 * Style properties for the container
+	 */
+	style?: React.CSSProperties;
+	/**
+	 * Size of the action icons
+	 */
+	size?: number;
+	/**
+	 * Icon color
+	 */
+	color?: string;
+	/**
+	 * Icon color when disabled
+	 */
+	colorDisabled?: string;
+	/**
+	 * Button variant
+	 */
+	variant?: string;
+	/**
+	 * Button variant when disabled
+	 */
+	variantDisabled?: string;
+	/**
+	 * Style for individual icons
+	 */
+	iconStyle?: MantineStyleProp;
+	/**
+	 * Optional list of session IDs to check for pending parameter changes.
+	 * If provided, buttons will be disabled when there are pending changes for these sessions.
+	 */
+	sessionIds?: string[];
```

Comment:
@mathieu-shapediver do you think we will need this (disable history buttons if there are pending parameter changes)? I would avoid it, because the browser history buttons have the exact same functionality and can't be disabled as well.

Reply:
@mathieu-shapediver ok sounds good, let's keep it! 
Note: The browser history buttons can't be blocked, let's keep the current behavior.